File: app/src/main/java/com/mintthursday/recipelist/RecipeListFragment.java
package com.mintthursday.recipelist;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.mintthursday.App;
import com.mintthursday.MainActivity;
import com.mintthursday.Router;
import com.mintthursday.recipe.creation.NewRecipeFragment;
import com.mintthursday.R;
import com.mintthursday.models.Recipe;
import com.mintthursday.recipe.creation.ingredientcreation.IngredientFragment;
import com.mintthursday.recipe.show.ShowRecipeFragment;

import java.util.List;

public class RecipeListFragment extends Fragment implements View.OnClickListener {

    private androidx.appcompat.view.ActionMode actionMode;

    private RecipeAdapter recipeAdapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_recipe_list, container, false);
        Toolbar myToolbar = view.findViewById(R.id.myToolbar);
        ((AppCompatActivity)getActivity()).setSupportActionBar(myToolbar);
        FloatingActionButton fab = view.findViewById(R.id.fab1);
        fab.setOnClickListener(this);

        initRecyclerViewMain(view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        loadRecipes();
    }

    private void initRecyclerViewMain(View view) {
        RecyclerView recipeRecyclerView = view.findViewById(R.id.recipe_recycler_view_main);
        recipeRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recipeAdapter = new RecipeAdapter();
        recipeRecyclerView.setAdapter(recipeAdapter);
        OnRecipeClickListener a = (new OnRecipeClickListener() {
            @Override
            public void onItemClick(Recipe recipe, int itemPosition) {
                if(getActivity() instanceof MainActivity) {
                    ((Router) getActivity()).showRecipe(recipe);
                }
            }

            @Override
            public boolean onItemLongClick(Recipe recipe, int position) {
                if (actionMode != null) {
                    return false;
                }
                actionMode =  ((AppCompatActivity) getActivity()).startSupportActionMode(new androidx.appcompat.view.ActionMode.Callback() {
                    @Override
                    public boolean onCreateActionMode(androidx.appcompat.view.ActionMode mode, Menu menu) {
                        mode.getMenuInflater().inflate(R.menu.menu_main_activity, menu);
                        mode.setTitle("Hi");
                        return true;
                    }

                    @Override
                    public boolean onPrepareActionMode(androidx.appcompat.view.ActionMode mode, Menu menu) {
                        return false;
                    }

                    @Override
                    public boolean onActionItemClicked(androidx.appcompat.view.ActionMode mode, MenuItem item) {
                        switch (item.getItemId()) {
                            case R.id.correct_recipe:
                                ((Router) getActivity()).editRecipe(recipe);
                                mode.finish();
                                return true;
                            case R.id.delete_recipe:
                                App.getInstance().getDatabase().recipeDao().delete(recipe);
                                loadRecipes();
                                mode.finish();
                                return true;
                            default:
                                return false;
                        }
                    }

                    @Override
                    public void onDestroyActionMode(androidx.appcompat.view.ActionMode mode) {
                        actionMode = null;
                    }
                });
                return true;

            }
        });
        recipeAdapter.setOnItemClickListenerRecipe(a);

    }

    private void loadRecipes() {
        List<Recipe> recipes = App.getInstance().getDatabase().recipeDao().getAll();
        recipeAdapter.setItems(recipes);
    }

    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.fab1:
                ((Router) getActivity()).createNewRecipe();
                break;
            default:
                break;
        }
    }
}

File: app/src/main/java/com/mintthursday/recipelist/RecipeListFragment.kt
package com.mintthursday.recipelist

import android.os.Bundle
import android.view.*
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.view.ActionMode
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.navigation.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import com.mintthursday.App.Companion.instance
import com.mintthursday.MainActivity
import com.mintthursday.R
import com.mintthursday.databinding.FragmentRecipeListBinding
import com.mintthursday.models.Recipe
import com.mintthursday.recipe.creation.NewRecipeFragment
import com.mintthursday.recipe.show.ShowRecipeFragment

class RecipeListFragment : Fragment() {

    private var _binding: FragmentRecipeListBinding? = null
    private val binding get() = _binding!!

    private var actionMode: ActionMode? = null
    private lateinit var recipeAdapter: RecipeAdapter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentRecipeListBinding.inflate(inflater, container, false)

        (activity as? AppCompatActivity)?.setSupportActionBar(binding.myToolbar)
//        binding.fab.setOnClickListener { instance.router.navigateTo(Screens.createNewRecipe()) }
        binding.fab.setOnClickListener { binding.root.findNavController().navigate(R.id.action_recipeListFragment_to_newRecipeFragment) }
        initRecyclerViewMain(binding)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        loadRecipes()
    }

    private fun initRecyclerViewMain(binding: FragmentRecipeListBinding) {
        binding.recipeRecyclerViewMain.layoutManager = LinearLayoutManager(context)
        recipeAdapter = RecipeAdapter()
        binding.recipeRecyclerViewMain.adapter = recipeAdapter
        recipeAdapter.setOnItemClickListenerRecipe(object : OnRecipeClickListener {
            override fun onItemClick(recipe: Recipe, position: Int) {
                if (activity is MainActivity) {
//                    instance.router.navigateTo(showRecipe(recipe))
                    binding.root.findNavController().navigate(R.id.action_recipeListFragment_to_showRecipeFragment, bundleOf(ShowRecipeFragment.ARG_RECIPE to recipe))
                }
            }

            override fun onItemLongClick(recipe: Recipe, position: Int): Boolean {
                if (actionMode != null) {
                    return false
                }
                actionMode = (activity as? AppCompatActivity)?.startSupportActionMode(object : ActionMode.Callback {
                    override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
                        mode.menuInflater.inflate(R.menu.menu_main_activity, menu)
                        mode.title = "Hi"
                        return true
                    }

                    override fun onPrepareActionMode(mode: ActionMode, menu: Menu): Boolean {
                        return false
                    }

                    override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
                        return when (item.itemId) {
                            R.id.correct_recipe -> {
//                                instance.router.navigateTo(editRecipe(recipe))
                                binding.root.findNavController().navigate(R.id.action_recipeListFragment_to_newRecipeFragment, bundleOf(NewRecipeFragment.ARG_EDIT_RECIPE to recipe))
                                mode.finish()
                                true
                            }
                            R.id.delete_recipe -> {
                                instance.database.recipeDao().delete(recipe)
                                loadRecipes()
                                mode.finish()
                                true
                            }
                            else -> false
                        }
                    }

                    override fun onDestroyActionMode(mode: ActionMode) {
                        actionMode = null
                    }
                })
                return true
            }
        })
    }

    private fun loadRecipes() {
        val recipes = instance.database.recipeDao().getAll()
        recipeAdapter.setItems(recipes)
    }
}